requirements.txt

```diff
@@ -1 +0,0 @@
-python_prctl
```

Comment:
And this will appear again if #1221 is acceptable? ...

Reply:
Indeed, #1221 is now showing as conflicting on requirements.txt and setup.py. Could you rebase once this PR is in and use an optional dependency in setup.py rather than requirements.txt please?

Reply:
The method with pip you've proposed will work only for pip and incompatible with plain setuptools. What should I write into `tests_require` in that case?

Reply:
I'm not sure what you mean by not compatible with setuptools. We can still do `python setup.py install` and our package is installed. It still has no dependencies. If we want to depend on `prctl` we could list it in install_requires.

So `prctl` is an optional component. Maybe it could be in an extras_require category like 'enhancements' and that be strongly recommended when we check the dependencies? Bear in mind the task here was to extend our existing dependency checking to include OS-specific system packages dependencies of optional dependencies.

What is your suggestion for an alternative approach, maybe pipenv? I've read some more about python packaging (there seems to be an infinite amount of it!) but it all suggests that `pip install .[category]` after `python setup.py install` is the way to go. `pip` deals with installing a concrete environment rather than an abstract dependency and altering os thread naming by installing an optional package sounds environmental to me.

As for requirements for tests, I have no opinion. Either we can have `tests_require` specify the abstract dependencies so that `python setup.py test` installs the test deps and runs the tests, or we can have an extras_require named 'test' and do something like `pip install .[test]; pytest`. Or we can do both - define the list at the top of setup.py and use it in both places.